site/fs.js

```diff
@@ -0,0 +1,49 @@
+const fs = require('fs').promises
+const path = require('path')
+const rimraf = require('rimraf')
+const util = require('util')
+
+const rimrafAsync = util.promisify(rimraf)
+
+const copyDirRecursiveAsync = async (src, dest) => {
+  try {
+    await fs.mkdir(dest, { recursive: true })
+  } catch (_) {
+    // ignore errors for mkdir
+  }
+
+  const childrenItems = await fs.readdir(src)
+
+  await Promise.all(
+    childrenItems.map(async item => {
+      const srcPath = path.join(src, item)
+      const destPath = path.join(dest, item)
+
+      const itemStat = await fs.lstat(srcPath)
+
+      if (itemStat.isFile()) {
+        fs.copyFile(srcPath, destPath)
+      } else {
+        await copyDirRecursiveAsync(srcPath, destPath)
+      }
+    })
+  )
+}
+
+const ensureFilePathAsync = async filePath => {
```

Comment:
We probably should let consumers pass the `dirname` directly instead.
Like this, functions like `copyDirRecursiveAsync()` will be able to use it.

Reply:
changed to `ensurePathExistsAsync = async path => {`